pkg/scheduler/extender/extender.go

```diff
@@ -589,27 +609,69 @@ func nodePrioritize(nodeMapping map[string][]volcrd.Volume) (map[string]int64, i
 	return nrank, maxCount
 }
 
-// scNameStorageTypeMapping reads k8s storage class resources and collect map with key storage class name
-// and value .parameters.storageType for that sc, collect only sc that have provisioner e.provisioner
-func (e *Extender) scNameStorageTypeMapping(ctx context.Context) (map[string]string, error) {
-	scs := storageV1.StorageClassList{}
+func generateEphemeralVolumeName(podName, volumeName string) string {
+	return podName + "-" + volumeName
+}
+
+// scChecker keeps info about the related SCs (provisioned by CSI Baremetal) and
+// the unrelated ones (prvisioned by other CSI drivers)
+type scChecker struct {
+	relatedSCs   map[string]string
+	unrelatedSCs map[string]bool
+}
+
+// buildSCChecker creates an instance of scChecker
+func (e *Extender) buildSCChecker(ctx context.Context, log *logrus.Entry) (*scChecker, error) {
+	ll := log.WithFields(logrus.Fields{
+		"method": "buildSCChecker",
+	})
+
+	var (
+		result = &scChecker{relatedSCs: map[string]string{}, unrelatedSCs: map[string]bool{}}
+		scs    = storageV1.StorageClassList{}
+	)
 
 	if err := e.k8sCache.ReadList(ctx, &scs); err != nil {
 		return nil, err
 	}
 
-	scNameTypeMap := map[string]string{}
 	for _, sc := range scs.Items {
 		if sc.Provisioner == e.provisioner {
-			scNameTypeMap[sc.Name] = strings.ToUpper(sc.Parameters[base.StorageTypeKey])
+			result.relatedSCs[sc.Name] = strings.ToUpper(sc.Parameters[base.StorageTypeKey])
+		} else {
+			result.unrelatedSCs[sc.Name] = true
 		}
 	}
-	if len(scNameTypeMap) == 0 {
+
+	ll.Debugf("related SCs: %+v", result.relatedSCs)
+	ll.Debugf("unrelated SCs: %+v", result.unrelatedSCs)
+
+	if len(result.relatedSCs) == 0 {
 		return nil, fmt.Errorf("there are no any storage classes with provisioner %s", e.provisioner)
 	}
-	return scNameTypeMap, nil
+
+	return result, nil
 }
 
-func generateEphemeralVolumeName(podName, volumeName string) string {
-	return podName + "-" + volumeName
+// scChecker.check return codes
+const (
+	relatedSC = iota
```

Comment:
May be names like `managed` \ `unmanaged` are better. `related` \ `unrelated` sound confusing to me.

Reply:
Fixed